src/rpdk/core/contract/suite/handler_commons.py

```diff
@@ -154,8 +154,15 @@ def test_input_equals_output(resource_client, input_model, output_model):
     pruned_output_model = prune_properties_if_not_exist_in_path(
         output_model.copy(), pruned_input_model, resource_client.read_only_paths
     )
+
     pruned_output_model = prune_properties_if_not_exist_in_path(
         pruned_output_model, pruned_input_model, resource_client.create_only_paths
     )
-
-    assert pruned_input_model == pruned_output_model
+    # only comparing properties in input model to those in output model and
+    # ignoring extraneous properties that maybe present in output model.
+    trunc_output_model = {}
+    for key in pruned_input_model:
+        if key in pruned_output_model:
+            trunc_output_model[key] = pruned_output_model[key]
+
+    assert pruned_input_model == trunc_output_model
```

Comment:
can we rewrite this a bit?

```python
assert all([pruned_input_model[key] == pruned_output_model[key] for key in pruned_input_model if key in pruned_output_model]), "Some alerting details why it failed"
```

Reply:
Can I use same error message as contract_asserts::response_contains_resource_model_equal_current_model ?
>All properties specified in the request MUST be present in the model  returned, and they MUST match exactly, with the exception of properties defined as writeOnlyProperties in the resource schema

Reply:
not sure how ambiguous that message is for this comparison; also DeepDiff seems to be a bit overkill for this usecase